contrib/parseq-tracevis-server/src/main/java/com/linkedin/parseq/GraphvizEngine.java

```diff
@@ -0,0 +1,185 @@
+package com.linkedin.parseq;
+
+import com.linkedin.parseq.exec.Exec;
+import com.linkedin.parseq.function.Success;
+import com.linkedin.parseq.function.Tuple2;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import javax.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class GraphvizEngine {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GraphvizEngine.class);
+
+  private final String _dotLocation;
+  private final Path _cacheLocation;
+  private final long _timeoutMs;
+  private final HashManager _hashManager;
+  private final Exec _exec;
+  private final ConcurrentHashMap<String, Task<Exec.Result>> _inFlightBuildTasks;
+
+  public GraphvizEngine(final String dotLocation, final Path cacheLocation, final int cacheSize, final long timeoutMs,
+      final int numThreads, final long reaperDelayMs, final int processQueueSize) {
+    _dotLocation = dotLocation;
+    _cacheLocation = cacheLocation;
+    _timeoutMs = timeoutMs;
+    _hashManager = new HashManager(this::removeCached, cacheSize);
+    _exec = new Exec(numThreads, reaperDelayMs, processQueueSize);
+    _inFlightBuildTasks = new ConcurrentHashMap<>();
+  }
+
+  public void start() {
+    _exec.start();
+  }
+
+  public void stop() {
+    _exec.stop();
+  }
+
+  /**
+   * Return task that has general HTTP status and body information based on the build task's result.
+   */
+  public Task<Tuple2<Integer, String>> build(final String hash, final InputStream body)
```

Comment:
It would be better to use a class that has fields: status and body instead of Tuple.